pkg/urbit/vere/io/ames.c

```diff
@@ -435,6 +550,112 @@ _ames_hear_bail(u3_ovum* egg_u, u3_noun lud)
   u3_ovum_free(egg_u);
 }
 
+/* _ames_put_packet(): add packet to queue, drop old packets on pressure
+*/
+static void
+_ames_put_packet(u3_ames* sam_u,
+                 u3_noun  msg,
+                 u3_noun  lan)
+{
+  u3_noun wir = u3nc(c3__ames, u3_nul);
+  u3_noun cad = u3nt(c3__hear, u3nc(c3n, lan), msg);  // trnasfer, unchanged
+
+  u3_auto_peer(
+    u3_auto_plan(&sam_u->car_u,
+                 u3_ovum_init(0, c3__a, wir, cad)),
+    0, 0, _ames_hear_bail);
+
+  _ames_cap_queue(sam_u);
+}
+
+/*  _ames_forward(): forward pac_u onto lan, then free pac_u
+*/
+static void
+_ames_forward(u3_panc* pac_u, u3_noun lan)
+{
+  c3_o nal = c3n;
+
+  //  repack the maybe-updated packet
+  //
+  {
+    //  unpack (jam [(unit lane) body])
+    //
+    u3_noun lon, bod;
+    {
+      u3_noun old = u3ke_cue(pac_u->bod_u.con);
+      u3x_cell(old, &lon, &bod);
+      u3k(lon); u3k(bod);
+      u3z(old);
+    }
+
+    //  only replace the lane if it was ~
+    //
+    if (u3_nul == lon) {
+      u3z(lon);
+      lon = u3nt(u3_nul, c3n, u3k(pac_u->ore));
+      //TODO  ore is always opaque lane. what about the sender=galaxy case?
```

Comment:
ames.hoon ignores galaxy ip/port lanes that vere sends it, so I'm not sure it matters